src/cc/tests/distributed_test.cc

```diff
@@ -610,6 +610,214 @@ TEST(DistributedTest, FullNeighborsMultipleServers)
+        auto partition = TestGraph::convert(path.path, "0_0", std::move(test_graphs[server]), 1);
+        servers.emplace_back(std::make_shared<snark::GRPCServer>(
+            std::make_shared<snark::GraphEngineServiceImpl>(path.string(), std::vector<uint32_t>{0},
+                                                            snark::PartitionStorageType::memory, ""),
+            std::shared_ptr<snark::GraphSamplerServiceImpl>{}, "localhost:0", "", "", ""));
+        channels.emplace_back(servers.back()->InProcessChannel());
+    }
+
+    return std::make_pair(std::move(servers), std::make_shared<snark::GRPCClient>(std::move(channels), 1, 1));
+}
+
+TEST(DistributedTest, NodeTypesMultipleTypesNeighborsSpreadAcrossPartitions)
+{
+    std::vector<std::vector<float>> f0 = {std::vector<float>{1.0f, 2.0f, 3.0f}};
+    std::vector<std::vector<float>> f1 = {std::vector<float>{4.0f, 5.0f, 6.0f}};
+    std::vector<std::vector<float>> f2 = {std::vector<float>{7.0f, 8.0f, 9.0f}};
+
+    auto environment =
+        CreateMultiServerSplitFeaturesEnvironment("NodeTypesMultipleTypesNeighborsSpreadAcrossPartitions", f0, f1, f2);
+    auto &c = *environment.second;
+
+    std::vector<snark::NodeId> nodes = {0, 1, 2};
+    std::vector<snark::Type> types(3, -3);
+
+    c.GetNodeType(std::span(nodes), std::span(types), -2);
+    EXPECT_EQ(std::vector<snark::Type>({0, 1, 2}), types);
+}
+
+TEST(DistributedTest, NodeFeaturesMultipleTypesNeighborsSpreadAcrossPartitions)
+{
+    std::vector<std::vector<float>> f0 = {std::vector<float>{1.0f, 2.0f, 3.0f}};
+    std::vector<std::vector<float>> f1 = {std::vector<float>{4.0f, 5.0f, 6.0f}};
+    std::vector<std::vector<float>> f2 = {std::vector<float>{7.0f, 8.0f, 9.0f}};
+
+    auto environment = CreateMultiServerSplitFeaturesEnvironment(
+        "NodeFeaturesMultipleTypesNeighborsSpreadAcrossPartitions", f0, f1, f2);
+    auto &c = *environment.second;
+
+    // 0 is a normal node
+    // 1, 2 has a parity with type = -1
+    // 3 is non existant
+    std::vector<snark::NodeId> nodes = {0, 1, 2, 3};
+    std::vector<uint8_t> output(4 * 3 * 4);
+    std::vector<snark::FeatureMeta> features = {{0, 12}};
+
+    c.GetNodeFeature(std::span(nodes), std::span(features), std::span(output));
+    std::span res(reinterpret_cast<float *>(output.data()), output.size() / sizeof(float));
+    EXPECT_EQ(std::vector<float>(std::begin(res), std::end(res)),
+              std::vector<float>({1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 0, 0}));
+}
+
+TEST(DistributedTest, NodeStringFeaturesMultipleTypesNeighborsSpreadAcrossPartitions)
+{
+    std::vector<std::vector<float>> f0 = {std::vector<float>{1.0f, 2.0f, 3.0f}};
+    std::vector<std::vector<float>> f1 = {std::vector<float>{4.0f, 5.0f, 6.0f}};
+    std::vector<std::vector<float>> f2 = {std::vector<float>{7.0f, 8.0f, 9.0f}};
+
+    auto environment = CreateMultiServerSplitFeaturesEnvironment(
+        "NodeFeaturesMultipleTypesNeighborsSpreadAcrossPartitions", f0, f1, f2);
```

Comment:
nit: use test name as environment name to store binary data in separate folders, `NodeStringFeaturesMultipleTypesNeighborsSpreadAcrossPartitions`, same below.

Reply:
same below